src/Util.h

```diff
@@ -472,8 +483,8 @@ std::unique_ptr<A> clone(const std::unique_ptr<A>& node) {
 }
 
 template <typename A>
-std::vector<std::unique_ptr<A>> clone(const std::vector<std::unique_ptr<A>>& xs) {
-    std::vector<std::unique_ptr<A>> ys;
+auto clone(const std::vector<A>& xs) {
```

Comment:
While you're here, the variable names could be a lot more meaningful. `xs` ->`nodes` and `ys` -> `newNodes` maybe?

Reply:
Technically `A` doesn't need to be AST nodes. `souffle::clone` could have been more verbosely named `souffle::deepCopy`. (Right now there's some non-kosher things going on with the `std::unique_ptr<A>` instances where they blindly assume there's a `A::clone` to call, but that could/should be overloadable.)